### PR TITLE
Add documentation for implementing new compiler passes.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title. Also fixes up a few British English spellings.

### CODE REVIEW

**Comment by sklam** [comment]
Can be merged once CI passes

File: docs/source/developer/compiler_pass_example.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        _name = "consts_add_one" # the common name for the pass
+
+        def __init__(self):
+            FunctionPass.__init__(self)
+
+        # implement method to do the work, "state" is the internal compiler
+        # state from the CompilerBase instance.
+        def run_pass(self, state):
+            func_ir = state.func_ir # get the FunctionIR object
+            mutated = False # used record whether this pass mutates the IR
```
Comment (esc) [inline docs/source/developer/compiler_pass_example.py]: a `to` is missing?

Comment (stuartarchibald) [reply docs/source/developer/compiler_pass_example.py]: thanks, fixed.

**Comment by esc** [comment]
For azure only the coverage run timed out. Probably safe to merge.

**Pull request merged**